FatPkg/EnhancedFatDxe/DiskCache.c

```diff
@@ -139,15 +342,15 @@ FatExchangeCachePage (
     //
     // Only fat table writing will execute more than once
     //
-    Status = FatDiskIo (Volume, IoMode, EntryPos, RealSize, PageAddress, Task);
+    Status = CacheFatDiskIo (CacheTag, DataType, Volume, IoMode, EntryPos, RealSize, PageAddress, Task);
     if (EFI_ERROR (Status)) {
       return Status;
     }
 
     EntryPos += Volume->FatSize;
   } while (--WriteCount > 0);
 
-  CacheTag->Dirty    = FALSE;
+  SetMem (CacheTag->DirtyBlocks, sizeof (CacheTag->DirtyBlocks), 0);

```

Comment:
This should be ZeroMem()

Reply:
Might also consider adding an internal static API to clear the cache, to make it clear where in this algorithm the cache state is cleared and when it is marked as dirty.

Reply:
Sure, I can add that for clarity.